distributed/worker_state_machine.py

```diff
@@ -3518,30 +3518,30 @@ def validate_state(self) -> None:
         for worker, tss in self.data_needed.items():
             for ts in tss:
                 fetch_tss.add(ts)
-                assert ts.state == "fetch"
-                assert worker in ts.who_has
+                assert ts.state == "fetch", self.story(ts)
+                assert worker in ts.who_has, f"{ts}; {ts.who_has=}"
```

Comment:
Nice, I didn't know about the `=` self-documenting expression in f-strings! (Documented [here](https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging) for anybody else who hasn't seen this before.)